lib/Driver/ToolChains.cpp

```diff
@@ -1287,6 +1287,26 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  StringRef Clang = "clang++";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_toolchain)) {
```

Comment:
It sounds from the Darwin toolchain comments like you might prefer to use `clang` as a driver for `ld` anyway. It's unfortunate that everything is so rigid and hard-coded on Darwin, but that shouldn't hold other platforms back, really.